sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -475,6 +475,7 @@ def create_item(
         :param pre_trigger_include: trigger id to be used as pre operation trigger.
         :param post_trigger_include: trigger id to be used as post operation trigger.
         :param indexing_directive: Indicate whether the document should be omitted from indexing.
+        :keyword bool disable_automatic_id_generation: Enable automatic id generation if no id present.
```

Comment:
The default value is True right? If we want non-breaking behaviour, we will probably need to keep automatic generation as disabled by default. So I think we could do that one of two ways:
- Keep the parameter called `disable_auto_id_gen`, and document the default as being `True` (and set it as such if not present).
- Rename the parameter to `enable_auto_id_gen`, document the default as being `False`, and set the request option to the opposite.

Reply:
@zfoster - on thinking about this a bit more - I think we should use the latter option - as I believe it conveys the behaviour more intuitively. i.e. update the kwarg name to `enable_auto_id_generation`, and update the docstring to state explicitly that this is `False` by default.
It would mean also updating the implementation to the opposite:
```
request_options["disableAutomaticIdGeneration"] = not kwargs.pop('disable_auto_id_gen', True)
```
Could we please also add a test (or update an existing test) to make sure that this new flag is tested?